### PR TITLE
Default warden client update

### PR DESCRIPTION
**Pull request by jbhatt-salesforce**

### CODE REVIEW

File: ArgusCore/src/test/java/com/salesforce/dva/argus/service/AlertServiceIT.java [hunk 1]
```diff
@@ -99,7 +99,9 @@ public void testExecuteScheduledAlerts_ForOneTimeSeries() throws InterruptedExce
             for (int i = 0; i < expectedNotifications; i++) {
                 for (int j = 0; j < 1 + random.nextInt(5); j++) {
                     alertService.enqueueAlerts(Arrays.asList(new Alert[] { alert }));
+                    sleep(5000);
```
Comment (saaja-sfdc) [inline ArgusCore/src/test/java/com/salesforce/dva/argus/service/AlertServiceIT.java:102]: @jbhatt-salesforce Do these delays really need to be this long?  This one test will take minutes to run.  Generally unit tests should run extremely quickly.

File: WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java [hunk 2]
```diff
@@ -1,5 +1,305 @@
...
+    private String metricName;
+    private TriggerType triggerType;
+    private Aggregator aggregator;
+    private List<Double> threshold;
+    private String timeUnit;
+    private Double defaultValue;
+    private String cronEntry;
+    private List<BigInteger> suspensionLevelIds = new ArrayList<BigInteger>(  );
+
+    public String getServcie(  )
```
Comment (saaja-sfdc) [inline WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java]: @jbhatt-salesforce Misspelled method name.  Should be getService()

File: WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java [hunk 3]
```diff
@@ -1,5 +1,305 @@
...
+    {
+        return name;
+    }
+
+    public void setName( String name )
+    {
+        this.name = name;
+    }
+
+    public List<String> getOwner(  )
```
Comment (saaja-sfdc) [inline WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java]: @jbhatt-salesforce Should be getOwners() plural.

File: WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java [hunk 4]
```diff
@@ -1,5 +1,305 @@
...
+    {
+        return owners;
+    }
+
+    public void setOwner( List<String> owner )
```
Comment (saaja-sfdc) [inline WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java]: @jbhatt-salesforce Should be setOwners() plural.

File: WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java [hunk 5]
```diff
@@ -1,5 +1,305 @@
...
+    {
+        this.owners = owner;
+    }
+
+    public List<String> getUser(  )
```
Comment (saaja-sfdc) [inline WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java]: @jbhatt-salesforce The following two methods should be plural form.  getUsers() and setUsers().

File: WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java [hunk 6]
```diff
@@ -1,5 +1,305 @@
...
+    {
+        this.metricName = metricName;
+    }
+
+    public TriggerType getTriggerType(  )
+    {
+        return triggerType;
+    }
+
+    public void setTriggerType( String triggerType )
```
Comment (saaja-sfdc) [inline WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java]: @jbhatt-salesforce the setters for triggerType and aggregator use the string name as the parameter, but the getters return the actual object.  It should be consistent between the two I think.  Either both getter and setter use the object, or string, but it should be the same between the two.

File: WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java [hunk 7]
```diff
@@ -1,5 +1,305 @@
...
+    {
+        return aggregator;
+    }
+
+    public void setAggregator( String aggregator )
+    {
+        this.aggregator = Aggregator.fromString( aggregator );
+    }
+
+    public List<Double> getThreshold(  )
```
Comment (saaja-sfdc) [inline WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java]: @jbhatt-salesforce should be plural form.

File: WardenApi/src/main/java/com/salesforce/dva/warden/entity/Entity.java [hunk 8]
```diff
@@ -0,0 +1,5 @@
+package com.salesforce.dva.warden.entity;
```
Comment (saaja-sfdc) [inline WardenApi/src/main/java/com/salesforce/dva/warden/entity/Entity.java]: @jbhatt-salesforce Should this file be removed from the WardenApi project?

File: WardenApi/src/main/java/com/salesforce/dva/warden/entity/JPAEntity.java [hunk 9]
```diff
@@ -0,0 +1,5 @@
+package com.salesforce.dva.warden.entity;
```
Comment (saaja-sfdc) [inline WardenApi/src/main/java/com/salesforce/dva/warden/entity/JPAEntity.java]: @jbhatt-salesforce Should this file also be removed from the WardenAPI project?

File: WardenApi/src/main/java/com/salesforce/dva/warden/entity/SuspensionLevel.java [hunk 10]
```diff
@@ -0,0 +1,12 @@
+package com.salesforce.dva.warden.entity;
```
Comment (saaja-sfdc) [inline WardenApi/src/main/java/com/salesforce/dva/warden/entity/SuspensionLevel.java]: @jbhatt-salesforce Should this file be removed from the WardenAPI project?

File: ArgusCore/src/test/java/com/salesforce/dva/argus/service/AlertServiceIT.java (hunk 1 above)
Comment (jbhatt-salesforce) [inline ArgusCore/src/test/java/com/salesforce/dva/argus/service/AlertServiceIT.java:102]: @saaja-sfdc, I think this was added by @rajsarkapally-sfdc to make the integration tests pass.

File: WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java (hunk 2 above)
Comment (jbhatt-salesforce) [inline WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java]: fixed.

File: WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java (hunk 3 above)
Comment (jbhatt-salesforce) [inline WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java]: fixed

File: WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java (hunk 5 above)
Comment (jbhatt-salesforce) [inline WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java]: fixed

File: ArgusCore/src/test/java/com/salesforce/dva/argus/service/AlertServiceIT.java (hunk 1 above)
Comment (rajsarkapally) [inline ArgusCore/src/test/java/com/salesforce/dva/argus/service/AlertServiceIT.java:102]: @jbhatt-salesforce we do not run integration tests in git, so you can delete line 102.

File: WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java (hunk 6 above)
Comment (jbhatt-salesforce) [inline WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java]: fixed

File: WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java (hunk 7 above)
Comment (jbhatt-salesforce) [inline WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java]: fixed

File: WardenApi/src/main/java/com/salesforce/dva/warden/entity/SuspensionLevel.java (hunk 10 above)
Comment (jbhatt-salesforce) [inline WardenApi/src/main/java/com/salesforce/dva/warden/entity/SuspensionLevel.java]: Yes, i can remove these files once i've Ruofan's code on my local env. Its just a placeholder.

**Comment by saaja-sfdc** [comment]
@jbhatt-salesforce, I think this looks good!  Let's work together and get the tests passing and see if we can crank through some more web service methods.

**Comment by greatruofan** [comment]
Hi Jigna,

Just want to double check if we need Jason annotation for aggregator enum
in policy dto. Please ignore me if I am wrong.

Thanks/Ruofan

**Comment by greatruofan** [comment]
I think another option is to use @JsonFormat
not sure about it,  but maybe it is easier for object transform

**Comment by saaja-sfdc** [comment]
@jbhatt-salesforce let's resolve these conflicts and merge this today.

**Pull request closed without merging**